homeassistant/components/apsystems_local/sensor.py

```diff
@@ -0,0 +1,166 @@
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production",
+        translation_key="today_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1 + c.e2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p1",
+        translation_key="today_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p2",
+        translation_key="today_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = config["COORDINATOR"]
+    device_name = config[CONF_NAME]
+    device_id = config[CONF_DEVICE_ID]
+
+    add_entities(
+        ApSystemsSensorWithDescription(coordinator, desc, device_name, device_id)
+        for desc in SENSORS
+    )
+
+
+class ApSystemsSensorWithDescription(CoordinatorEntity, SensorEntity):
+    """Base sensor to be used with description."""
+
+    entity_description: ApsystemsLocalApiSensorDescription
+
+    def __init__(
+        self,
+        coordinator: ApSystemsDataCoordinator,
+        entity_description: ApsystemsLocalApiSensorDescription,
+        device_name: str,
+        device_id: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._attr_native_value: int | float | None = None
+        self.entity_description = entity_description
+        self._device_name = device_name
+        self._device_id = device_id
+        self._attr_unique_id = f"apsystems_local_{device_id}_{entity_description.key}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Get the DeviceInfo."""
+        return DeviceInfo(
+            identifiers={("apsystems_local", self._device_id)},
+            name=self._device_name,
+            serial_number=self._device_id,
+            manufacturer="APsystems",
+            model="EZ1-M",
+        )
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if self.coordinator.data is not None:
+            self._attr_native_value = self.entity_description.value_fn(
+                self.coordinator.data
+            )
+        self.async_write_ha_state()
```

Comment:
Shouldn't we only write the state if the value has updated?
```suggestion
            self.async_write_ha_state()
```